pkg/api/v1/routes/handlers.go

```diff
@@ -163,8 +168,74 @@ func (r *Routes) serverConditionCreate(c *gin.Context) (int, *v1types.ServerResp
 		}
 	}
 
+	return r.conditionCreate(otelCtx, newCondition, serverID, facilityCode)
+}
+
+func (r *Routes) serverEnroll(c *gin.Context) (int, *v1types.ServerResponse) {
+	id := c.Param("uuid")
+	otelCtx, span := otel.Tracer(pkgName).Start(c.Request.Context(), "Routes.serverEnroll")
+	span.SetAttributes(attribute.KeyValue{Key: "serverId", Value: attribute.StringValue(id)})
+	defer span.End()
+
+	var conditionCreate v1types.ConditionCreate
+	if err := c.ShouldBindJSON(&conditionCreate); err != nil {
+		r.logger.WithFields(logrus.Fields{
+			"error": err,
+		}).Info("invalid ConditionCreate payload")
+
+		return http.StatusBadRequest, &v1types.ServerResponse{
+			Message: "invalid ConditionCreate payload: " + err.Error(),
+		}
+	}
+
+	var serverID uuid.UUID
+	var err error
+	if id != "" {
+		serverID, err = uuid.Parse(id)
+		if err != nil {
+			r.logger.WithFields(logrus.Fields{
+				"serverID": id,
+			}).Info("bad serverID")
+
+			return http.StatusBadRequest, &v1types.ServerResponse{
+				Message: err.Error(),
+			}
+		}
+	} else {
+		serverID = uuid.New()
+	}
+
+	var params v1types.AddServerParams
+	if err = json.Unmarshal(conditionCreate.Parameters, &params); err != nil {
+		return http.StatusBadRequest, &v1types.ServerResponse{
+			Message: "invalid params: " + err.Error(),
+		}
+	}
+
+	// Creates a server record in FleetDB.
+	if err = r.fleetDBClient.AddServer(c.Request.Context(), serverID, params.Facility, params.IP, params.Username, params.Password); err != nil {
+		if strings.Contains(err.Error(), badRequestErrMsg) {
+			return http.StatusBadRequest, &v1types.ServerResponse{
+				Message: err.Error(),
+			}
+		}
+		return http.StatusInternalServerError, &v1types.ServerResponse{
+			Message: err.Error(),
+		}
+	}
+
+	// Reuse the inventory for server enrollment.
+	// May change in the future.
+	kind := rctypes.Kind(rctypes.Inventory)
+
+	newCondition := conditionCreate.NewCondition(kind)
```

Comment:
At this point, the `Parameters` field of newCondition is going to be the payload we bound at line 181, right? So you're creating an `Inventory` condition with `AddServer` parameters. What we need to do here is create a [InventoryTaskParameters](https://github.com/metal-toolbox/rivets/blob/e4c2952b8cb4530334d7a4135f0dc994376375e6/condition/inventory.go#L21) and serialize that. This would be something like:
```
inventoryArgs := rctypes.{
    CollectBiosConfig: true,
    CollectFirmwareStatus: true,
    Method: rctypes.Outofband,
    AssetID: serverID,
}
``` 
then serialize that to JSON and set it as the `Parameters` in newCondition.

Reply:
Thanks!
The current version filtered out params for creating the server. PTAL thanks!